src/painting/painter.rs

```diff
@@ -209,7 +217,7 @@ impl Painter {
         }
 
         // Lines and distance parameters
-        let remaining_lines = self.remaining_lines();
+        let remaining_lines = self.remaining_lines() + self.prompt_height;
```

Comment:
Edge case: not enough space to even draw the prompt fully, in which case `remaining_lines` should be 0, but will be positive with this change. Might be safer to use `self.screen_height().saturating_sub(self.prompt_start_row)` directly